sources/engine/Stride.Graphics/BufferPool.cs

```diff
@@ -35,8 +35,11 @@ internal BufferPool(GraphicsResourceAllocator allocator, GraphicsDevice graphics
             this.allocator = allocator;
 
             Size = size;
+
+#pragma warning disable 162 // Unreachable code detected
```

Comment:
The compiler only sees one value for UseBufferOffsets so will complain about unreachable code, depending on the value.   Encapsulating the code branches where UseBufferOffsets is used with warning disable/enable is the simplest way of addressing these.